addons/product/models/product_template.py

```diff
@@ -415,6 +416,30 @@ def _get_related_fields_variant_template(self):
         """ Return a list of fields present on template and variants models and that are related"""
         return ['barcode', 'default_code', 'standard_price', 'volume', 'weight', 'packaging_ids']
 
+    def _get_base_or_overridden_price(self, price_type):
+        """Allow other modules to override the price during the computation of the final price (based on the rules of
+        the price list, ...).
+
+        This method overrides the list_price if record_being_sold is set by getting the price (field price) from the
+        record targeted by record_being_sold on the model identified by the detailed_type and the mapping
+        DETAILED_TYPE_MODELS. The price is assumed to be in product currency.
+        Other price types are not overridden (ex.: standard_price).
+
+        See example in event module where we need to compute the final price for different tickets with different price
+        but linked to the same product.
+
+        :returns tuple(price, is_overridden):
+            price (float): the base price if is_overridden is False otherwise the overridden price
+            is_overridden (bool): whether the price is overridden or the base price
```

Comment:
Better parsed by IDE (at least ours), and avoids repetition:
```py
"""
        :returns: price: the base price if is_overridden is False otherwise the overridden price.  
                  is_overridden: whether the price is overridden or the base price
        :rtype: (float, bool)
""" 
```

Reply:
Thanks for the tip